test/general/signature.js

```diff
@@ -1046,36 +1020,18 @@ bwM=
   });
 
   it('Verify fails with signed message with critical notations', async function() {
-    let testFailed = true;
-    try {
-      openpgp.config.tolerant = false;
-      const sMsg = await openpgp.readMessage({ armoredMessage: signature_with_critical_notation });
-      const pub_key = await openpgp.readKey({ armoredKey: pub_key_arm2 });
-      const verified = await sMsg.verify([pub_key]);
-      await verified[0].verified;
-      testFailed = false;
-    } catch (e) {
-      expect(e.message).to.equal('Unknown critical notation: test@example.com');
-    } finally {
-      openpgp.config.tolerant = true;
-    }
-    // fail the test if execution does not throw an exception
-    expect(testFailed).to.be.true;
+    const message = await openpgp.readMessage({ armoredMessage: signature_with_critical_notation });
+    const key = await openpgp.readKey({ armoredKey: pub_key_arm2 });
+    const { signatures: [sig] } = await openpgp.verify({ message, publicKeys: key });
+    expect(sig.error).to.match(/Unknown critical notation: test@example.com/);
```

Comment:
Just to be extra cautious, maybe add
```suggestion
    expect(sig.valid).to.be.false;
    expect(sig.error).to.match(/Unknown critical notation: test@example.com/);
```